pygem/params.py

```diff
@@ -294,6 +299,89 @@ def print_info(self):
+			*** the trimmed part, however, is still saved inside a file. It is just 'invisible' when drawn in a program
+		:param float triangulate_tol: tolerance of triangulation (size of created triangles)
+		:return:
+		"""
+		min_xyz, max_xyz = self._calculate_bb_dimension(shape, tol, triangulate, triangulate_tol)
+		self.origin_box = min_xyz
+		self._set_box_dimensions(min_xyz, max_xyz)
+		self._set_position_of_vertices()
+		self._set_mapping()
+		self._set_transformation_params_to_zero()
+
+	def _set_box_dimensions(self, min_xyz, max_xyz):
+		"""
+		Dimensions of the cage are set as distance from the origin (minimum) of the cage to
+		the maximal point in each dimension.
+		:param iterable min_xyz: three values representing the minimal values of the bounding box in XYZ respectively
+		:param iterable max_xyz: three values representing the maximal values of the bounding box in XYZ respectively
+		:return:
+		"""
+		dims = [max_xyz[i] - min_xyz[i] for i in range(3)]
+		self.lenght_box_x = dims[0]
+		self.lenght_box_y = dims[1]
+		self.lenght_box_z = dims[2]
+
+	def _set_position_of_vertices(self):
+		"""
+		Vertices of the control box around the object are set in this method.
+		Four vertex (non coplanar) are sufficient to uniquely identify a parallelepiped -- the
+		second half of the box is created as a mirror reflection of the first four vertices.
+		:return:
+		"""
+		origin_array = np.array(self.origin_box)
+		dim = [self.lenght_box_x, self.lenght_box_y, self.lenght_box_z]
+		self.position_vertex_0 = origin_array
+		self.position_vertex_1 = origin_array + np.array([dim[0], .0, .0])
+		self.position_vertex_2 = origin_array + np.array([.0, dim[1], .0])
+		self.position_vertex_3 = origin_array + np.array([.0, .0, dim[2]])
+
+	def _set_mapping(self):
+		dim = [self.lenght_box_x, self.lenght_box_y, self.lenght_box_z]
+		self.psi_mapping = np.diag([1. / dim[i] for i in range(3)])
+		self.inv_psi_mapping = np.diag(dim)
+
+	def _set_transformation_params_to_zero(self):
+		ctrl_pnts = self.n_control_points
+		self.array_mu_x = np.zeros(ctrl_pnts)
+		self.array_mu_y = np.zeros(ctrl_pnts)
+		self.array_mu_z = np.zeros(ctrl_pnts)
+
+	def _calculate_bb_dimension(self, shape, tol=1e-6, triangulate=False, triangulate_tol=1e-1):
+		""" Calculate dimensions (minima and maxima) of a box bounding the given shape
+		----------
+		:param TopoDS_Shape shape: or a subclass such as TopoDS_Face
+			the shape to compute the bounding box from
+		:param float tol: tolerance of the computed bounding box
+		:param bool triangulate :
+			if True only the dimensions of the bb will take into account every part of the shape (also not 'visible')
+			if False only the 'visible' part is taken into account
+			*See: `build_bounding_box`
+		:param float triangulate_tol: tolerance of triangulation (size of created triangles)
+		:return tuple: consisting of two tuples: first one has coords of minimum, the second one coords of maximum
```

Comment:
use the :rtype: for the type and return for the name. See the last method of freeform.py to see an example